src/PhpParser/Parser/RectorParser.php

```diff
@@ -23,7 +35,22 @@ public function __construct(
      */
     public function parseFile(SmartFileInfo $smartFileInfo): array
     {
-        return $this->parser->parseFile($smartFileInfo->getRealPath());
+        try {
+            return $this->parser->parseFile($smartFileInfo->getRealPath());
+        } catch (ParserErrorsException $parserErrorsException) {
+            $message = $parserErrorsException->getMessage();
+
+            if (! StringUtils::isMatch($message, self::FN_ERROR_REGEX)) {
+                throw $parserErrorsException;
+            }
+
+            $parser = $this->parser;
+
+            $lexer = &\Closure::bind(static fn&($parser) => $parser->lexer, null, $parser)($parser);
+            $lexer = new Emulative(['phpVersion' => Emulative::PHP_7_3]);
```

Comment:
@TomasVotruba it possibly needs to update `Emulative` Lexer instead, I tried this, but got error:

```bash
There was 1 error:

1) Rector\Tests\Php74\Rector\Function_\ReservedFnFunctionRector\ReservedFnFunctionRectorTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
TypeError: PhpParser\Lexer::getTokens(): Return value must be of type array, null returned
```